examples/stable/test/property/canister_methods/on_low_wasm_memory/test/tests.ts

```diff
@@ -0,0 +1,176 @@
+}
+
+/**
+ * Validates that the error received matches the expected memory limit error
+ *
+ * @param error - The error object to validate
+ */
+function validateMemoryLimitError(error: unknown): void {
+    const canisterId = getCanisterId(CANISTER_NAME);
+    expect(error).toMatchObject({
+        name: 'AgentError',
+        methodName: 'addRandomBytes',
+        type: 'update',
+        reject_code: 5,
+        error_code: 'IC0539',
+        canisterId: Principal.from(canisterId),
+        reject_message: expect.stringMatching(
+            new RegExp(
+                `Error from Canister ${canisterId}: ` +
+                    'Canister exceeded its current Wasm memory limit of \\d+ bytes\\. ' +
+                    'The peak Wasm memory usage was \\d+ bytes\\. ' +
+                    'If the canister reaches 4GiB, then it may stop functioning and may become unrecoverable\\. ' +
+                    'Please reach out to the canister owner to investigate the reason for the increased memory usage\\. ' +
+                    'It might be necessary to move data from the Wasm memory to the stable memory\\. ' +
+                    'If such high Wasm memory usage is expected and safe, then the developer can increase the Wasm memory limit in the canister settings\\.\\.' +
+                    '\\nTry checking the canister for a possible memory leak or modifying it to use more stable memory instead of Wasm memory\\. ' +
+                    'See documentation: http://internetcomputer\\.org/docs/current/references/execution-errors#wasm-memory-limit-exceeded'
+            )
+        )
+    });
+}
+
+/**
+ * Validates the initial state of the canister:
+ * - Confirms low memory handler has not been called
+ * - Verifies wasm memory limit matches configured value
+ *
+ * @param actor - The canister actor instance
+ * @param wasmMemoryLimit - The configured wasm memory limit
+ */
+async function validateInitialStatus(
+    actor: Actor,
+    wasmMemoryLimit: number
+): Promise<void> {
+    const lowMemoryHandlerCalledAtBeginning =
+        await actor.wasLowMemoryHandlerCalled();
+
+    expect(lowMemoryHandlerCalledAtBeginning).toBe(false);
+
+    const initialStatus = getCanisterStatus(CANISTER_NAME);
+    expect(initialStatus.wasmMemoryLimit).toBe(wasmMemoryLimit);
+}
+
+/**
+ * Validates the final state of the canister after memory operations:
+ * - Verifies memory size exceeds (wasmMemoryLimit - wasmMemoryThreshold)
+ * - Verifies memory size exceeds wasmMemoryLimit (current behavior)
+ * - Confirms low memory handler was called
+ *
+ * @param actor - The canister actor instance
+ * @param wasmMemoryLimit - The configured wasm memory limit
+ * @param wasmMemoryThreshold - The configured memory threshold
+ */
+async function validateFinalStatus(
+    actor: Actor,
+    wasmMemoryLimit: number,
+    wasmMemoryThreshold: number
+): Promise<void> {
+    const finalStatus = getCanisterStatus(CANISTER_NAME);
+
+    // This is the behavior I am expecting
```

Comment:
Comment and link to issue/forum post with the true final expected behavior

Reply:
Done